internal/core/data/controller/http/event.go

```diff
@@ -23,24 +24,24 @@ import (
 )
 
 type EventController struct {
-	readers map[string]io.EventReader
+	readers map[string]io.DtoReader
 	dic     *di.Container
 }
 
 // NewEventController creates and initializes an EventController
 func NewEventController(dic *di.Container) *EventController {
 	return &EventController{
-		readers: make(map[string]io.EventReader),
+		readers: make(map[string]io.DtoReader),
 		dic:     dic,
 	}
 }
 
-func (ec *EventController) getReader(r *http.Request) io.EventReader {
+func (ec *EventController) getReader(r *http.Request) io.DtoReader {
```

Comment:
add RWLock for accessing reader from the map

Reply:
Added.